src/main/java/org/spongepowered/mod/registry/SpongeGameRegistry.java

```diff
@@ -279,6 +280,15 @@ public TradeOfferBuilder getTradeOfferBuilder() {
         return Optional.fromNullable((EntityType) this.entityIdToTypeMappings.get(id));
     }
 
+    public Optional<EntityType> getEntity(Class<? extends org.spongepowered.api.entity.Entity> entityClass) {
+        for(EntityType type : getEntities()) {
```

Comment:
@simon816 In my testing that didn't work correctly. Probably because the class object is talking about the MC object that the interface was mixed in to, and as we are just passing the interface, it doesn't notice it.